nx/lib/nx/lin_alg.ex

```diff
@@ -949,6 +949,44 @@ defmodule Nx.LinAlg do
     )
   end
 
+  @doc """
+  Calculates the Moore-Penrose inverse, or the pseudoinverse, of a matrix.
+
+  ## Options
+    * `:eps` - Rounding error threshold used to assume values as 0.
+
+  ## Examples
+
+      iex> pinv = Nx.LinAlg.pinv(Nx.tensor([[1, 1], [3, 4]]))
+      #Nx.Tensor<
+        f32[2][2]
+        [
+          [4.0, -1.0000001192092896],
+          [-3.0, 1.0000001192092896]
+        ]
+      >
+
+      iex> Nx.LinAlg.pinv(Nx.tensor([[1, 3], [5, 4], [4, 9]]))
+      #Nx.Tensor<
+        f32[2][3]
+        [
+          [-0.07312049716711044, 0.3027806580066681, -0.11019568890333176],
+          [0.06900102645158768, -0.1307930052280426, 0.1462409943342209]
+        ]
+      >
+  """
+  defn pinv(tensor, opts \\ []) do
```

Comment:
Thank you. Let’s rename the function to pseudo_inverse?

We also need to make sure it works batched. If you pass a higher than rank-2 tensor, we should assume that each additional dimension works as a batch. :)

Reply:
The only issue I have with `pseudo_inverse` is that for normal inverse we use `Nx.LinAlg.invert`. Should we change that to `inverse`?

Reply:
for more context, `np.linalg.pinv` is how it is done in numpy

Reply:
@polvalente should we call it `pseudo_invert` then? or `pinv`? What do you decide?

Reply:
pinv is a really usual name for this function

All of the following langs/libs use it: NumPy/Jax, PyTorch, Julia, MATLAB/Octave

So I think it's ok to go with pinv :)